js/popup.js

```diff
@@ -0,0 +1,176 @@
+              fill="white"
+            />
+          </g>
+        </g>
+        <defs>
+          <clippath id="clip0-48-2223">
+            <rect width="18" height="20" fill="white" transform="translate(3 2)" />
+          </clippath>
+          <clippath id="clip1-48-2223">
+            <rect width="18" height="20" fill="white" transform="translate(3 2)" />
+          </clippath>
+        </defs>
+      </svg>
+    </a>
+  </div>
+</div>
+</div>`;
+
+const updatePopupContent = (id) => {
+  const popupIndex = projectCardAttributes.findIndex((attr) => attr.id === id);
+  const newPopupAttributes = projectCardAttributes[popupIndex];
+
+  if (newPopupAttributes) {
+    const projectCard = generatePopupProjectCard(
+      newPopupAttributes.src,
+      newPopupAttributes.title,
+      newPopupAttributes.alt,
+      newPopupAttributes.mdDescription,
+      newPopupAttributes.tagLi,
+      newPopupAttributes.btnHrefFirst,
+      newPopupAttributes.btnHrefSecond,
+      newPopupAttributes.popupNavPrevious,
+      newPopupAttributes.popupNavNext,
+    );
+
+    popUpCardContainerNew.innerHTML = projectCard;
+
+    const closeBtn = document.querySelector('.popup-card-close-btn');
+    closeBtn.addEventListener('click', function closeBtnForPopup() {
+      popUpCardContainerNew.classList.add('d-none');
+      popUpCardContainerNew.classList.remove('d-flex');
+    });
+  }
+
+  const popupNavNext = document.querySelector('.popup-nav-next');
+  const popupNavPrevious = document.querySelector('.popup-nav-previous');
+
+  let popupIndexClick = projectCardAttributes.findIndex((attr) => attr.id === id);
+  const updateAndMove = (direction) => {
+    if (direction === 'next') {
+      popupIndexClick =
+        popupIndexClick < projectCardAttributes.length - 1 ? popupIndexClick + 1 : popupIndexClick;
+    } else if (direction === 'previous') {
+      popupIndexClick = popupIndexClick > 0 ? popupIndexClick - 1 : popupIndexClick;
+    }
+
+    updatePopupContent(projectCardAttributes[popupIndexClick].id);
+  };
+
+  if (popupIndexClick === 0) {
+    popupNavPrevious.setAttribute('disabled', 'true');
+  }
+
+  if (popupIndexClick === projectCardAttributes.length - 1) {
+    popupNavNext.setAttribute('disabled', 'true');
+  }
+
+  popupNavNext.addEventListener('click', () => updateAndMove('next'));
+  popupNavPrevious.addEventListener('click', () => updateAndMove('previous'));
```

Comment:
This might be redundant 

```suggestion
popUpCardContainerNew.addEventListener('click', handlePopupButtonClick)
```